packages/eyes/src/eyes-test/eyes.ts

```diff
@@ -0,0 +1,168 @@
+    };
+  }
+  return resources;
+}
+
+function logEyesResult({name, status, url, error}: IResult) {
+  const {compName, simName, styleName} = parseFilename(name, '.snapshot.html');
+
+  const isModified = status === 'modified';
+  const isNew = status === 'new';
+  const isError = status === 'error';
+
+  const formattedUrl = isModified ? `${chalk.cyan('URL')}: ${chalk.underline(url)}` : '';
+  const statusMessage = isModified ? chalk.red('MODIFIED') :
+               isNew ? chalk.green('NEW') :
+               isError ? chalk.bgRedBright('ERROR') :
+               chalk.green('UNMODIFIED');
+  consoleLog(`${statusMessage} ${chalk.bold(compName)}: ${simName}. Style: ${styleName}. ${formattedUrl}`);
+
+  if (isError) {
+    consoleLog(error);
+  }
+}
+
+// We cannot allow any of the result promises to reject, because we're going to
+// await on a bunch of them in parallel using Promise.all, and a single rejected
+// promise would reject the entire batch.
+function getTestResult(testName: string, testResult: Promise<any>): Promise<IResult> {
+  return (
+    testResult
+    .catch((err: any) => err)
+    .then((res: any) => res instanceof TestFailedError ? res.getTestResults() : res)
+    .then((res: any) => Array.isArray(res) && res[0] instanceof TestResults ? res[0] : res)
+    .then((res: any): IResult => {
+      const url = res.getUrl();
+
+      if (res instanceof TestResults) {
+        if (res.getIsDifferent()) {
+          return {name: testName, status: 'modified', url};
+        }
+        if (res.getIsNew()) {
+          return {name: testName, status: 'new', url};
+        }
+        if (res.isPassed()) {
+          return {name: testName, status: 'unmodified', url};
+        }
+      }
+
+      return {name: testName, status: 'error', url, error: res};
+    })
+  );
+}
+
+async function runTest(gridClient: any, gridClientConfig: any, testName: string, html: string, resources: {[url: string]: IResource}) {
+  const dom = new JSDOM(html).window.document;
+
+  const {checkWindow, close} = await gridClient.openEyes({
+    ...gridClientConfig,
+    testName,
+  });
+
+  await checkWindow({
+    url: 'http://localhost/html/index.html',
+    cdt: domNodesToCdt(dom),
+    sizeMode: 'viewport',
+    resourceContents: resources
+  });
+
+  return {testName, testResult: close()};
+}
+
+export async function runEyes(projectPath: string, directory: string) {
+  const cssFilenames  = glob.sync('*.css', {cwd: directory});
+  const htmlFilenames = glob.sync('*.snapshot.html', {cwd: directory});
+
+  const config = getGridClientConfig(projectPath);
+  const resources = getStaticResources(cssFilenames, directory);
+  const gridClient = makeVisualGridClient(initConfig());
+
+  const resultPromises = [];
+  consoleLog('Sending snapshots to Applitools...');
+
+  for (const htmlFilename of htmlFilenames) {
+    const html = fs.readFileSync(path.join(directory, htmlFilename), 'utf-8');
+    const {testName, testResult} = await runTest(
```

Comment:
Here we're passing htmlFilename as `testName` to `runTest` function, which returns it back to us, and we assign it to `testName`.
1) There's no reason to return testName from runTest, because we already have it.
2) Test name should not be the same as the filename. We're sending it to Applitools and that's the string users will see as the title of the test in the dashboard.